src/pex-builder/builder/util.py

```diff
@@ -0,0 +1,129 @@
+import json
+import logging
+import os
+import subprocess
+import sys
+from contextlib import contextmanager
+from typing import List, Tuple
+from zipfile import ZipFile
+
+import click
+from dagster_cloud_cli import gql
+
+logging.basicConfig(level=logging.INFO)
+
+
+def run_self_module(module_name, args: List[str]):
+    "Invoke this executable again with -m {module}"
+    # If running a pex file directly, we invoke the executable pex file again.
+    # Otherwise we assume we're running in a pex generated venv and use the python executable.
+    cmd = os.getenv("PEX", sys.executable)
+
+    args = [cmd, "-m", module_name] + args
+    logging.info("Running %r", args)
+    proc = subprocess.run(args, capture_output=True)
+    return proc
+
+
+def get_pex_flags(python_version: Tuple[str, str]) -> List[str]:
```

Comment:
Maybe https://packaging.pypa.io/en/latest/version.html#packaging.version.Version instead of rolling our own type?

Reply:
great suggested. switched.